src/test/java/teammates/ui/webapi/UpdateNotificationActionTest.java

```diff
@@ -0,0 +1,126 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.Const;
+import teammates.ui.output.NotificationData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+import teammates.ui.request.NotificationUpdateRequest;
+
+/**
+ * SUT: {@link UpdateNotificationAction}.
+ */
+public class UpdateNotificationActionTest extends BaseActionTest<UpdateNotificationAction> {
+    private static final String TEST_NOTIFICATION = "notification1";
+    private final NotificationAttributes testNotificationAttribute = typicalBundle.notifications.get(TEST_NOTIFICATION);
+
+    @Override
+    String getActionUri() {
+        return Const.ResourceURIs.NOTIFICATION;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return PUT;
+    }
+
+    @Test
+    @Override
+    protected void testExecute() throws Exception {
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_ID, testNotificationAttribute.getNotificationId(),
+        };
+        NotificationUpdateRequest req = getTypicalUpdateRequest();
+        NotificationStyle style = req.getStyle();
+        NotificationTargetUser targetUser = req.getTargetUser();
+        String title = req.getTitle();
+        String message = req.getMessage();
+        String invalidTitle = "";
+        String invalidNotificationId = "InvalidNotificationId";
+
+        loginAsAdmin();
+
+        ______TS("Typical Case: Update notification successfully");
+        req = getTypicalUpdateRequest();
+        UpdateNotificationAction action = getAction(req, requestParams);
+        NotificationData res = (NotificationData) action.execute().getOutput();
+
+        NotificationAttributes updatedNotification = logic.getNotification(res.getNotificationId());
+
+        assertEquals(res.getStartTimestamp(), updatedNotification.getStartTime().toEpochMilli());
+        assertEquals(res.getEndTimestamp(), updatedNotification.getEndTime().toEpochMilli());
```

Comment:
I feel these two lines are not consistent with the 4 assertions below.

`style`, `targetUser`... are values input and are "expected values". However, `res` and `updatedNotification` are both fetched from db/backend so they are "actual values". Not saying comparing the two "actual values" are wrong as they are fetched differently and it does verify that the returned value from API is the updated value in DB. However, to be consistent with the below assertions, comparison with `req.getStartTimestamp()` with `updatedNotification` is more preferred in my opinion.

Reply:
Have added the changes to my latest commit